modules/iam-assumable-role-with-oidc/main.tf

```
@@ -62,6 +62,28 @@ data "aws_iam_policy_document" "assume_role_with_oidc" {
   }
 }
 
+data "aws_iam_policy_document" "assume_self" {
+  count = var.explicit_permission_to_assume_self ? 1 : 0
+
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:role${var.role_path}${var.role_name}"]
```

For Govcloud and CN support:

```suggestion
      identifiers = ["arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:role${var.role_path}${var.role_name}"]
```